packages/hooks/src/usePaymentAgentTransferVisible.ts

```diff
@@ -5,7 +5,9 @@ const usePaymentAgentTransferVisible = () => {
     const { client } = useStore();
     const { is_authorize } = client;
 
-    const { data, ...rest } = useFetch('get_settings', { options: { enabled: is_authorize } });
+    const { data, ...rest } = useFetch('get_settings', {
+        options: { enabled: is_authorize, refetchOnWindowFocus: false },
```

Comment:
<img src="https://github.com/binary-com/deriv-app/assets/72082844/9f981c48-e8a0-4c00-9b28-2a529a42f028" width="300px" />

Reply:
It is fetching extra requests to the server :)

Reply:
I thought you were sick today, wanted to connect regarding this 😄

Reply:
> It is fetching extra requests to the server :)

But what's the problem with that 😅 `get_settings` is not subscribe-ale,  So by re-fetching it when it's needed we make sure the data is not outdated.

Reply:
I was checking the card description, Not sure why we have cards like this now 🤔 

Instead of reducing the API calls at this stage by patches, We should fix the API calls first, We are calling `get_settings` in many places (cause it's not subscribe-ale), Components, Stores, Utils...

The first step should be moving all of those calls to `useFetch` behind the abstraction, Once we have it all behind the abstraction then we can control the cashing or re-fetching strategy in a centralized place (`QueryClient`) for each endpoint based on the need to make improvements. (And once we have the SDK thing we just change the abstraction instead of the whole codebase).

It's like we want to improve the house cooling system without even building the house first 😅

cc: @markwylde-deriv @yashim-deriv

Reply:
<img width="1426" alt="image" src="https://github.com/binary-com/deriv-app/assets/129206554/b4d00af8-bb40-4449-a200-5476bb041796">

Yes, I agree with you @farzin-deriv , right now the way we are fetching the data from server with `WS` calls if we try to replace the code with our `useFetch` hook that would be a bad idea since it will give up the declarative approach of `react-query`. 

see the screenshot for example. here I am trying to replicate the same usage by using `refetch` and `mutateAsync` from our `useRequest` and `useFetch` hook. the problem is we should not be fetching the data like this example with `react-query` since it is giving up the idea of declarative data fetching with `useQuery` hook.

After playing around with this, I think the better approach would be to refactor for instance the `submitHandler` like this one in the screenshot (we have similar callback functions on several places) to remove these direct server requests inside these functions and move all the fetching and mutating logic to a hook

cc: @ali-hosseini-deriv

Reply:
@amir-deriv That's what I just said 😅 And we are doing this in the cashier package.

We can take this `submitHandler` as an example and jump on a call and do pair programming with devs involved with this thing if you guys like 🙇🏻